### PR TITLE
Updating StartAskingForBalance

### PR DESCRIPTION
**Pull request by ronaldsg20**
I've change the way addresses are got from HDwallets in
                the startAskingForBalance method, asking and using the
                xpub value to derive them.

### CODE REVIEW

**Review by alexjavabraz — APPROVED** [review]
LGTM

File: src/services/WalletService.ts [hunk 1]
```diff
@@ -201,4 +221,32 @@ export default abstract class WalletService {
         return addressListResponse;
       });
   }
+
+  private async getAccountsXpub(accountIdx: number): Promise<void> {
```
Comment (josedahlquist) [inline src/services/WalletService.ts:225]: a `get` method that doesn't return anything? I would change that to `set` if what it does is set the internal field

**Review by josedahlquist — COMMENTED** [review]
The logic seems good to me.
We talked about re thinking how we decide whether a wallet should use xpub derivation or address fetching.
Instead of using a hdWallet flag I think it may make sense to define strategies that are passed/defined on each concrete walletService.

**Review by josedahlquist — APPROVED** [review]
Quick question, do we have tests for WalletService?
I think it would make sense to test these new methods

**Pull request merged**